Binseop.Ko step7 crud

Binseop-Ko opened the pull request:
概要
Step 7. create crud for task

実装したこと
Rails研修のStep7,「タスクを登録・更新・削除できるようにしよう

テスト項目・エビデンス（場合によって）
list
<img width="529" alt="list" src="https://user-images.githubusercontent.com/97071306/154209172-9b0f6c95-d122-4ff7-b832-703fb83173f3.png">
create
<img width="1061" alt="create" src="https://user-images.githubusercontent.com/97071306/154209187-854d7e64-ea9b-4269-bf6a-8ba2a14041d9.png">
view / edit / delete
<img width="1045" alt="read,modify,delete" src="https://user-images.githubusercontent.com/97071306/154209182-11de250a-4282-4ba9-8416-0b650f41e9a6.png">

確認方法
docker-compose up -d
# DBの用意が終わってから　（docker-compose up が終わってからも、しばらく時間がかかります。seedでエラーになる場合は１分ほど待っててからもう一回実行してください）
docker-compose exec api rails db:seed
access http://127.0.0.1:3001

BinseopKo committed:
step7

task crud

MioArai reviewed (commented):
1. Dockerfileのruby バージョンのせいか以下のエラーが出てしまいました。
```diff
#  Dockerfile
- FROM ruby:2.6
+ FROM ruby:2.6.9
```
```
 => ERROR [ 9/10] RUN bundle install                                                                        0.8s 
------                                                                                                           
 > [ 9/10] RUN bundle install:
#12 0.766 Your Ruby version is 2.6.8, but your Gemfile specified 2.6.9
```


2. board_id = 1のデータがないようですね。Seed追加したらよいですかね？
![image](https://user-images.githubusercontent.com/90371497/154451523-e528bb15-ee11-42b9-aae6-230f454524da.png)

MioArai commented on myapp/app/controllers/api/board_controller.rb:
```diff
@@ -0,0 +1,30 @@
+    end
+
+  end
+end
```
`No newline at end of file` が出てますので、IDEにプラグインや設定入れておいた方が良さそうですね。

Binseop-Ko replied:
@MioArai 修正しました

MioArai commented on myapp/app/controllers/home_controller.rb:
```diff
@@ -0,0 +1,8 @@
+  def index
+    #todo check auth, set proper board_id
+    board_id = 1
+    redirect_to "/board/" + String(board_id)
```
[FYI] Pathヘルパーを使われることが多いかもです
https://railsguides.jp/routing.html#%E3%82%B3%E3%83%B3%E3%83%88%E3%83%AD%E3%83%BC%E3%83%A9%E3%81%AE%E5%90%8D%E5%89%8D%E7%A9%BA%E9%96%93%E3%81%A8%E3%83%AB%E3%83%BC%E3%83%86%E3%82%A3%E3%83%B3%E3%82%B0

Binseop-Ko replied:
@MioArai ありがとうございます。修正しました

BinseopKo committed:
[RB] db seed

BinseopKo committed:
[RB]

Use path helper

yuta-d-saito commented on myapp/app/controllers/api/task_controller.rb:
```diff
@@ -0,0 +1,62 @@
+    def update
+      task = Task.find(params[:id])
+
+      #todo: check auth
+      editableColumns = [:title, :status_id, :due_date, :contents, :priority_id]
+      
+      for column in Task::EDITABLE_FIELDS
+        if(params.has_key?(column))
+          task[column] = params[column]
+        end
+      end
```
ここも同様にストロングパラメーターを使うとシンプルになるかもです！

yuta-d-saito commented on myapp/app/controllers/api/task_controller.rb:
```diff
@@ -0,0 +1,62 @@
+      #todo: check write permission
+      task.board_id = params[:board_id]
+      
+      for column in Task::EDITABLE_FIELDS
```
このままでも良いですが、ストロングパラメーターを使うとシンプルになるかもです！
https://pikawaka.com/rails/permit

■例

def create
@task = Task.new(task_params)
@task.save
end

private

def task_params
params.require(:task).permit(:id, :name, :description......など)
end

Binseop-Ko replied:
@yuta-d-saito 簡単な方法があったんですね！ありがとうございます。修正しました。

yuta-d-saito commented on myapp/db/seeds.rb:
```diff
@@ -5,3 +5,23 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
```
👍

yuta-d-saito commented on myapp/config/routes.rb:
```diff
@@ -1,3 +1,12 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  get root :to => 'home#index'
```
👍

MioArai commented on myapp/app/views/board/show.html.erb:
```diff
@@ -0,0 +1,80 @@
+    </form>
+</div>
+
+<script>
```
ゴリゴリにJS使っててクオリティー高い 👍

MioArai commented on myapp/app/views/board/show.html.erb:
```diff
@@ -0,0 +1,80 @@
+        <div class="mb-3">
+            <label class="form-label">Priority</label>
+            <select class="form-select" id="edit_priority">
+                <option value="Major" >Major</option>
```
[Q]この辺りはDBからの取得値に置き換わる感じですかね？

Binseop-Ko replied:
@MioArai そうですね、最初スクショ用の画面を作った時の残りです。消しました！

MioArai reviewed (approved):
![image](https://user-images.githubusercontent.com/90371497/154899578-c4392f22-883a-437b-a169-a965eedfdfad.png)

BinseopKo committed:
[RB]

use string parameters
fix bugs (create, status tab sorting)
remove unnecessary select options

yuta-d-saito reviewed (approved):
LGTM!
